src/environment/mutex.rs

```diff
@@ -0,0 +1,236 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+//! Internal mutex interface.
+
+use std::collections::HashMap;
+use std::num::NonZeroU32;
+
+use crate::libc::errno::{EBUSY, EDEADLK, EPERM};
+use crate::libc::pthread::mutex::HostMutexId;
```

Comment:
Hmm, wouldn't it make more sense for the definition of `HostMutexId` to be here rather than in pthreads, now? It could also just be called `MutexId` in that case; the host/guest distinction only matters within pthreads, which deals with both.